src/pages/StocktakeEditPage.js

```diff
@@ -51,130 +52,159 @@ import { gotoStocktakeManagePage } from '../navigation/actions';
+        if (propName === 'onCheckAction') return PageActions.selectRow;
+        return PageActions.deselectRow;
       case 'mostUsedReasonTitle':
-        return rowKey => openModal(STOCKTAKE_REASON, rowKey);
+        return onEditReason;
       default:
         return null;
     }
   });
 
+  const getModalOnSelect = () => {
+    switch (modalKey) {
+      case MODAL_KEYS.STOCKTAKE_COMMENT_EDIT:
+        return onEditComment;
+      case MODAL_KEYS.EDIT_STOCKTAKE_BATCH:
+        return onConfirmBatchEdit;
+      case MODAL_KEYS.STOCKTAKE_OUTDATED_ITEM:
+        return onResetStocktake;
+      case MODAL_KEYS.ENFORCE_STOCKTAKE_REASON:
+      case MODAL_KEYS.STOCKTAKE_REASON:
+        return onApplyReason;
+      default:
+        return null;
+    }
+  };
+
   const renderRow = useCallback(
     listItem => {
       const { item, index } = listItem;
       const rowKey = keyExtractor(item);
-      const { row, alternateRow } = newDataTableStyles;
       return (
         <DataTableRow
           rowData={data[index]}
           rowState={dataState.get(rowKey)}
           rowKey={rowKey}
-          style={index % 2 === 0 ? alternateRow : row}
           columns={columns}
           isFinalised={isFinalised}
           dispatch={dispatch}
-          focusCellAction={focusCell}
-          focusNextAction={focusNext}
           getAction={getAction}
+          rowIndex={index}
         />
       );
     },
     [data, dataState]
   );
 
-  const renderHeader = () => (
-    <DataTableHeaderRow
-      columns={columns}
-      dispatch={instantDebouncedDispatch}
-      sortAction={sortData}
-      isAscending={isAscending}
-      sortBy={sortBy}
-    />
+  const renderHeader = useCallback(
+    () => (
+      <DataTableHeaderRow
+        columns={columns}
+        dispatch={instantDebouncedDispatch}
+        sortAction={PageActions.sortData}
+        isAscending={isAscending}
+        sortBy={sortBy}
+      />
+    ),
+    [sortBy, isAscending]
   );
 
-  const getModalOnSelect = () => {
-    switch (modalKey) {
-      case STOCKTAKE_COMMENT_EDIT:
-        return value => dispatch(editComment(value, 'Stocktake'));
-      case EDIT_STOCKTAKE_BATCH:
-        return () => dispatch(closeStocktakeBatchModal());
-      case STOCKTAKE_OUTDATED_ITEM:
-        return () => runWithLoadingIndicator(() => dispatch(resetStocktake()));
-      case ENFORCE_STOCKTAKE_REASON:
-      case STOCKTAKE_REASON:
-        return ({ item }) => dispatch(applyReason(item));
-      default:
-        return null;
-    }
-  };
+  const PageButtons = useCallback(() => {
+    const ManageStocktake = () => (
+      <PageButton
+        text={buttonStrings.manage_stocktake}
+        onPress={onManageStocktake}
+        isDisabled={isFinalised}
+      />
+    );
+
+    // Program stocktakes do not have a ManageStocktake button.
+    const Child = program ? () => null : ManageStocktake;
+
+    return (
+      <View style={newPageTopRightSectionContainer}>
+        <Child />
```

Comment:
I think `{ !program && <ManageStocktake/> }` is better

Reply:
Agree 👍 .